text/0148-sampling-probability.md

```diff
@@ -0,0 +1,794 @@
+count of spans in the population using the spans that were sampled.
+The adjusted count associated with a Span is the expected value of the
+number of identical Spans within the population that each Span
+represents.  Probability samplers SHOULD ensure the samples they
+compute are unbiased, which implies that the sum
+of adjusted counts in the sample approximates the true count of spans in the
+population.
+
+The adjusted count in an unbiased probability sampling scheme SHOULD
+be expressed using a Span attribute named `sampling.adjusted_count`
+when it represents the whole population of events.  Adjusted count 
+attributes MAY be integer or floating-point values.
+
+#### Inclusion probability tracestate value
+
+The recommended way to convey sampling probability *for contexts* in
+OpenTelemetry is through the W3C Trace Context tracestate using the
+key `head_probability`.
+
+Probability Samplers MAY encode the effective sampling inclusion
+probability using tracestate, for the context that was in effect when
+the W3C is-sampled bit was set.  The tracestate field SHOULD be set in
+both sampled and unsampled cases, to convey the inclusion probability
+even for unsampled contexts.
+
+The `head_probability` tracestate key is set to a floating-point
+number greater than or equal to 0 and less than or equal to 1.  The
+floating point precision of the number SHOULD follow implementation
+language standards and SHOULD be high enough to identify when Samplers
+have different inclusion probabilities.
+
+#### Counting probabilistically sampled spans
+
+Consumers of a stream of span data that may or may not have been
+sampled can follow these steps to count or approximately count the
+total number of spans in the population.
+
+For each span processed, locate the `sampling.adjusted_count` attribute. 
+If there is none, count a single span event.  If the attribute is set, 
+count that many identical span events.
+```
+
+For the `TraceIDRatio` sampler, include the following additional text:
+
+```md
+When returning a `RECORD_AND_SAMPLE` decision, the TraceIDRatio
+Sampler MUST include the attribute `sampling.adjusted_count=C`, where
+`C` is the reciprocal of the configured trace ID ratio.
+
+The returned tracestate used for the child context MAY have the
+tracestate `otel` key with the sub-key `headprob` set to the configured 
+trace ID ratio.
+```
+
+For the `Parent` sampler, include the following additional text:
+
+```md
+When returning a `RECORD_AND_SAMPLE` decision, the Parent Sampler MAY
+include the attribute `sampling.adjusted_count=C`, where `C` is the
+reciprocal of the parent trace context's head inclusion probability.
+
+The tracestate `otel` key with the sub-key `headprob` is used to lookup
+and propagate the configured  trace ID ratio.  When the `otel` key with 
+sub-key `headprob` is not located and `is-sampled` is set, the Sampler 
+MUST set `sampling.adjusted_count=0` to signal that spans cannot be 
```

Comment:
How would setting it to 0 be different from not setting at all?

If we look at the existing trace context formats like Zipkin's B3 or Jaeger's, they do not propagate probability and they use classic head-based sampling with sampling decision propagated. Extrapolation from individual spans is not possible with those, since there's no way of knowing probability or count of children spans. Extrapolation from re-assembled trace is possible as long as the count is captured in the root. I think whichever behavior we propose, it should degrade nicely into this legacy (and very widespread) mode.

Reply:
The text is outdated, resolving.